### ISSUE TITLE
Closed Captioning Request

### ISSUE THREAD

**Issue by FoxbitDreamtail** [mwpenny/portal64-still-alive#82]
#### Version of the game 
 - Source Build: Aug.10.2024

#### What is the enhancement?
change the subtitles tick boxes to a drop-down menu and have the slider for text language only appear when subtitles are active

#### Screenshots
![image](https://github.com/user-attachments/assets/0a1d2eac-94c8-424b-9ad2-cc3315215579)

#### Additional context
as-is it takes up a lot of unnecessary space in the menu and(Potentially) 2 menu options for subtitles could be confusing as to their meaning and use.

**Comment by hackgrid** [comment]
No, dropdown menu would be needed to be implemented from scratch and does
not work well on N64 resolution.

Text language also affects menus without subtitles, it always needs to be
visible. (Except only one language was built-in)

Foxbit ***@***.***> schrieb am So., 11. Aug. 2024, 23:42:

**Comment by FoxbitDreamtail** [comment]
I redacted the language thing for a reason. I don't see a problen with the drop-down menu with 3 options, Disabled, Subtitles(Dialog Only), and Closed Captions. also, isn't this setting normally in 'Audio' in the main game?

**Comment by mwpenny** [comment]
A drop-down is possible and I think it _could_ potentially look okay (use same font as the menu itself), but it's non-negligible work for a pretty small benefit. And selecting the dropdown would result in effectively the same menu being displayed that we have today, but hidden behind one extra button press.

I don't mind keeping this open in case a dropdown is needed for some other menu element, or if someone wants to prototype it, but it will be quite low on the list.

You're right that it's normally in "audio". We have less space for menus at this low resolution, and it doesn't fit there. It's really a matter of perspective. After all, captions appear visually :)

**Comment by FoxbitDreamtail** [comment]
> You're right that it's normally in "audio". We have less space for menus at this low resolution, and it doesn't fit there. It's really a matter of perspective. After all, captions appear visually :)

What about adding scrolling in the menus? it would make chapter select a little more pleasant and closer to the Console versions of portal too!

**Comment by mwpenny** [comment]
Scrolling already exists for the control bindings menu, but that one's a bit different (just a list of selectable rows). The other menus don't really lend themselves well to it, particularly with the sliders. It's also easier for players if they can see everything at once where possible.

If more menu entries get added to the point of requiring scrolling outside of the controls menu, it can be tackled at that time. The menus would essentially need to be redone.

For the new game menu, console versions use a similar UI but also include arrows to indicate that scrolling is possible. Here is the Switch version:
![image](https://github.com/user-attachments/assets/aed71487-6025-4d76-a901-c1621fe003fb).
If this is what you mean, I agree it would be an improvement (and could also apply to the save/load menu). It isn't hard to add but is a low priority right now (PRs welcome).

**Comment by FoxbitDreamtail** [comment]
> For the new game menu, console versions use a similar UI but also include arrows to indicate that scrolling is possible. Here is the Switch version: ![image](https://private-user-images.githubusercontent.com/9356790/358439943-aed71487-6025-4d76-a901-c1621fe003fb.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjM3NzkyMTksIm5iZiI6MTcyMzc3ODkxOSwicGF0aCI6Ii85MzU2NzkwLzM1ODQzOTk0My1hZWQ3MTQ4Ny02MDI1LTRkNzYtYTkwMS1jMTYyMWZlMDAzZmIucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDgxNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA4MTZUMDMyODM5WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NzZiMTgyYWJhMzI1ZmQ1NjkwNWFjMmNlOWQyZDZiYmEzOTlkZTU5YWJlZjliN2VmZjk2MmNjMDgzMzZjYWJiZiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.Bhy109xoSF8wR6QQslzuWK2qKdeA2iTIMxQv0Rca6S4). If this is what you mean, I agree it would be an improvement (and could also apply to the save/load menu). It isn't hard to add but is a low priority right now (PRs welcome).

Yeah, that's what I meant, all the console versions look the same, the XB360 version of Portal has an identical menu.
